src/pages/ShopEditPage.tsx

```diff
@@ -1,3 +1,272 @@
+  const navigate = useNavigate();
+  const { user } = useUserStore();
+  const shopId = user?.shopId;
+  const fileInputRef = useRef<HTMLInputElement | null>(null);
+  const [imagePreview, setImagePreview] = useState<string | null>(null);
+  const [imageFile, setImageFile] = useState<File | null>(null);
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const [form, setForm] = useState<FormType>({
+    name: "",
+    category: "" as ShopCategory,
+    address1: "" as SeoulDistrict,
+    address2: "",
+    originalHourlyPay: "",
+    description: "",
+  });
+
+  useEffect(() => {
+    async function fetchShop() {
+      if (!shopId) return;
+      const res = await getShop(shopId);
+      const shop = res.data.item;
+      setForm({
+        name: shop.name,
+        category: shop.category,
+        address1: shop.address1,
+        address2: shop.address2,
+        originalHourlyPay: numberCommaFormatter(shop.originalHourlyPay),
+        description: shop.description,
+      });
+      if (shop.imageUrl) setImagePreview(shop.imageUrl);
+    }
+    fetchShop();
+  }, [shopId]);
+
+  const handleChange = (
+    key: keyof FormType,
+    value: string | SeoulDistrict | ShopCategory,
+  ) => {
+    setForm((prev) => ({ ...prev, [key]: value }));
+  };
+
+  const handleImageUpload = (e: ChangeEvent<HTMLInputElement>) => {
+    const file = e.target.files?.[0];
+    if (!file) return;
+
+    setImageFile(file);
+
+    const reader = new FileReader();
+    reader.onload = () => {
+      if (typeof reader.result === "string") {
+        setImagePreview(reader.result);
+      }
+    };
+    reader.readAsDataURL(file);
+  };
+
+  const handleSubmit = async () => {
+    if (isSubmitting || !shopId) return;
+
+    const requiredFields: Array<keyof typeof form> = [
+      "name",
+      "category",
+      "address1",
+      "address2",
+      "originalHourlyPay",
+    ];
+
+    const missingField = requiredFields.find((key) => form[key].trim() === "");
+
+    if (missingField) {
+      alert(`${FIELD_LABELS[missingField]}을(를) 입력해 주세요.`);
+      return;
+    }
+
+    const hourlyPay = Number(extractDigits(form.originalHourlyPay));
+    if (isNaN(hourlyPay) || hourlyPay <= 0) {
+      alert("유효한 시급을 입력해 주세요.");
+      return;
+    }
+
+    try {
+      setIsSubmitting(true);
+
+      let imageUrl = imagePreview || "";
+      if (imageFile) {
+        const presignedURL = await postImage(imageFile.name);
+        await putImage(presignedURL, imageFile);
+        imageUrl = getPublicURL(presignedURL);
+      }
+
+      const payload = {
+        name: form.name.trim(),
+        category: form.category,
+        address1: form.address1,
+        address2: form.address2.trim(),
+        originalHourlyPay: hourlyPay,
+        description: form.description.trim(),
+        imageUrl,
+      };
+
+      await putShop(shopId, payload);
+      navigate("/shop");
+    } finally {
+      setIsSubmitting(false);
+    }
```

Comment:
하나의 의견입니다! 😅

기존에 작성해주신 코드는
비동기 요청과 관련된 변수들을 가까이 두시려는 의도로 보이는데요 🤔

에러 발생가능성이 높은 비동기 요청과
에러 발생 가능성이 없는 변수를 번갈아봐야 해서
조금 읽기가 어려울 수 있을 것 같아요.

```suggestion
      setIsSubmitting(true);

      let imageUrl = imagePreview || "";

      const payload = {
        name: form.name.trim(),
        category: form.category,
        address1: form.address1,
        address2: form.address2.trim(),
        originalHourlyPay: hourlyPay,
        description: form.description.trim(),
        imageUrl,
      };

    try {
      if (imageFile) {
        const presignedURL = await postImage(imageFile.name);
        await putImage(presignedURL, imageFile);
        imageUrl = getPublicURL(presignedURL);
      }

      await putShop(shopId, payload);
      navigate("/shop");
    } finally {
      setIsSubmitting(false);
    }
```